### ISSUE TITLE
[Accessibility] [New feature request] Allow `Card` to accept `aria` props for rendered link

### ISSUE THREAD

**Issue by rfearing** [primer/brand#1398]
### Problem

The `Card` component accepts `href` and `ctaText` as props but there isn't any API for allowing `aria` props on the link. This opens up opportunities for inaccessible links. For example, an editor could pass in "Read more" to the `ctaText`. In Contentful, links are provided to the edotr with the props:
- href
- text
- alt

While it is encouraged to fill in descriptive text in the alt field, there is nothing, programmatically, from stopping "Learn more" or "Read more" or for the same text to be used multiple times on a page (all of which would constitute an a11y violation).

### Proposal

I suggest a prop such as `linkAria` or `linkAlt` be added to allow for an `aria-label` to exist.

[Related comment](https://github.com/github/github-ui/pull/26563#discussion_r3574290902)
